mozjs/src/rust.rs

```diff
@@ -433,6 +443,35 @@ impl Drop for Runtime {
     }
 }
 
+/// JSContext that is Send and Sync
+/// but supports only limited (thread safe) operations
+pub struct SyncJSContext(Arc<Mutex<Option<*mut JSContext>>>);
+
+unsafe impl Send for SyncJSContext {}
+unsafe impl Sync for SyncJSContext {}
+
+// We expose only methods that are defined as thread safe in spidermonkey
+// https://searchfox.org/mozilla-central/rev/7a85a111b5f42cdc07f438e36f9597c4c6dc1d48/js/src/vm/JSContext.h#848
```

Comment:
I think it makes sense to remove this comment in exchange for the rustdoc above.